gui/src/ScenarioViewer.tsx

```diff
@@ -0,0 +1,539 @@
+
+const tagsAutocompleteSx = {
+    "& .MuiOutlinedInput-root": {
+        padding: "3px 15px 3px 3px !important",
+    },
+    maxWidth: "none",
+};
+
+const PipelineRow = ({ action, number, value }: PipelinesRowProps) => {
+    const [pipeline, setPipeline] = useState<string>(value);
+    const [focus, setFocus] = useState(false);
+
+    const onFieldBlur = useCallback((e: any) => {
+        setPipeline(e.target.value);
+        setFocus(false);
+    }, []);
+
+    const index = number + 1;
+    return (
+        <Grid item xs={12} container justifyContent="space-between">
+            <Grid item container xs={10}>
+                {action === "EDIT" && (
+                    <TextField
+                        data-name={`pipe${index}`}
+                        label={"Pipeline " + index}
+                        variant="outlined"
+                        name={`pipeline${index}`}
+                        value={pipeline}
+                        onBlur={onFieldBlur}
+                        sx={FieldNoMaxWidth}
+                        onFocus={(e) => setFocus(true)}
+                        fullWidth
+                        InputProps={{
+                            endAdornment: focus && (
+                                <>
+                                    <IconButton sx={IconPaddingSx}>
+                                        <CheckCircle color="primary" />
+                                    </IconButton>
+                                    <IconButton sx={IconPaddingSx}>
+                                        <Cancel color="inherit" />
+                                    </IconButton>
+                                </>
+                            ),
+                        }}
+                    />
+                )}
+                {action !== "EDIT" && <Typography variant="subtitle2">{pipeline}</Typography>}
+            </Grid>
+            <Grid item xs={2} container alignContent="center" alignItems="center" justifyContent="center">
+                <IconButton component="label" size="small">
+                    <Send color="info" />
+                </IconButton>
+            </Grid>
+        </Grid>
+    );
+};
+
+const ScenarioViewer = (props: ScenarioViewerProps) => {
+    const { id = "", scenario = ["", false, "", "", "", [], [], [], []] } = props;
+
+    const [confirmDialogOpen, setConfirmDialogOpen] = useState(false);
+    const [properties, setProperties] = useState<Property[]>([]);
+    const [action, setAction] = useState<string>("EDIT");
+    const [tags, setTags] = useState<string[]>();
+    const [newProp, setNewProp] = useState<Property>({
+        id: "",
+        key: "",
+        value: "",
+    });
+    const [label, setLabel] = useState<string>();
+
+    const [labelFocus, setLabelFocus] = useState(false);
+    const [tagsFocus, setTagsFocus] = useState(false);
+
+    const sendScenario = useCallback((e: React.MouseEvent<HTMLElement>) => {}, []);
```

Comment:
rename to submitEntity
cf. scenarioSelector on how to dispatch an action (onSubmit with and id that would be attached to the button ad data-id)

Reply:
and pass to the pipeline for submission